### DISCUSSION TITLE
Manipulasi data di dataframe R

### DISCUSSION THREAD

**Discussion by fxpeterbs31** [teamalgoritma/community#243] [Q&A] [answered]
Halo, saya mau tanya.
<img width="45" alt="2022-08-03 00_41_51-LBB - RStudio" src="https://user-images.githubusercontent.com/107398327/182439398-1a037650-31e5-4cbe-91eb-a0724f6aa58f.png">
Pada kolom `size`, data yang saya miliki masih berupa string dengan format `<numeric><"k"/"M">` atau `Varies with device`. 
1. Dari format pertama, jika saya mau mengubah menjadi misal: 8.3M = 8300000 atau 2.5k = 2500 apakah memungkinkan? 
2. Dari format kedua, jika saya mau mengubah menjadi semacam `NaN` di python apakah memungkinkan? 

Terima kasih :)

**Comment by amaliasf** [comment, marked as answer]
Dear Bapak @fxpeterbs31 ,

1. Terkait pengubahan 8.3M = 8300000 atau 2.5k = 2500 dapat dilakukan dengan memanfaatkan fungsi `str_sub` dari library `stringr`. Untuk referensi code nya bisa merujuk pada pertanyaan berikut: https://github.com/teamalgoritma/community/discussions/224
2. Kemudian untuk mengubah suatu nilai menjadi `NaN` bisa menggunakan `naniar::replace_with_na()`. Berikut referensi terkait: https://cran.r-project.org/web/packages/naniar/vignettes/replace-with-na.html

Semoga membantu Pak, apabila sudah terjawab boleh di Mark as Answer :)
